xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/migration/AbstractDocumentsMigration.java

```diff
@@ -65,9 +62,7 @@ protected void hibernateMigrate() throws DataMigrationException
     {
         List<XWikiDocument> documents = selectDocuments()
             .stream()
-            .map(this::convert)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
+            .map(documentReference -> new XWikiDocument(this.documentReferenceResolver.resolve(documentReference)))
```

Comment:
Not sure if this change might deserve its own issue.

Reply:
Well I think what is needed is a much bigger change than that actually as loading all the impacted XWikiDocument before handling them it's quite horrible for the memory. It should instead, for each entry of `selectDocuments()`, load the  XWikiDocument and call `logBeforeQueuingTask`.

Reply:
https://jira.xwiki.org/browse/XWIKI-21089 created for this, I'll revert here and open a dedicated PR.